### DISCUSSION TITLE
How to hook to repeater create or delete item action?

### DISCUSSION THREAD

**Discussion by studiowizjo** [filamentphp/filament#1602] [Help] [answered]
I found in documentation this:

```php
protected function setUp(): void
{
    parent::setUp();
 
    $this->registerListeners([
        'repeater::createItem' => [
            function (Component $component, string $statePath): void {
                if ($component->isDisabled()) {
                    return;
                }
 
                if ($statePath !== $component->getStatePath()) {
                    return;
                }
 
                // ...
            },
        ],
    ]);
}
```

but it doesnt work if placed in Resource page or CreateRecord page. I suspect that I could extend Repeater field class and there put this code, but is there a better way to hook to this from resource pages?

**Comment by danharrin** [comment, marked as answer]
you can chain `->registerListeners()` onto your actual repeater component I think?

**Comment by studiowizjo** [comment]
Silly me :), but if so I think documentation is confusing and suggest rather extending field component than registering on run time, so maybe it is worth to extend this topic?

Offcourse, it works, and here is the solution for future generations:

```php
Repeater::make('expenses')
	->registerListeners([
		'repeater::deleteItem' => [
			function (Repeater $component, string $statePath, string $uuidToDelete): void {
				$component->getLivewire()->calculateExpensesAmount();
			},
		],
	])
```

but one things is additionaly worth mentioning: 

Registering this way adds second delete listener, so you are not overwriting the default one and don`t have to delete row by yourself.
